### PR TITLE
Form validation

### LINKED ISSUE #64: Server sided form validation
> Validation of data entered in HTML forms. Validate in, at least, view and integration layers on the server. The task only concerns the server side, you're not required to do any validation on the client side.

Higher Grade Score: 1p

### PR DESCRIPTION
**Pull request by WilliamRagstad**
Implemented an advanced thorough object validation service for server sided HTML form validation.

Closes #64

### CODE REVIEW

**Review by Willgar — APPROVED** [review]
Was there no previous working validation library available?
Looks good otherwise.

File: recruitment-api/service/ValidationService.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            `Unknown validation rule type ${(rule as any).type} for ${
+              JSON.stringify(data)
+            } (${typeof data})`,
+          );
+          return { isValid: false, errors: ["Unknown validation rule type"] };
+      }
+      return { isValid: true, casted: wasCasted ? castedValue : undefined };
+    }
+
+    public validate(
```
Comment (Willgar) [inline recruitment-api/service/ValidationService.ts:225]: Same here, documentation

**Review by Willgar — CHANGES_REQUESTED** [review]
Add a bit more documentation, then it should be good.

**Review by Luddeeeeeee — APPROVED** [review]

**Comment by WilliamRagstad** [comment]
> Was there no previous working validation library available?

Not that deeply validated an object structure against a typed schema, that is what is issued in this pull request. Another library I found ([deno_validator](https://deno.land/x/deno_validator)) could be used to check so that the fields have valid values. Might bee something to look into.

**Pull request merged**